Tests/StoreKitUnitTests/PurchasesOrchestratorTests.swift

```diff
@@ -426,6 +427,32 @@ class PurchasesOrchestratorTests: StoreKitConfigTestCase {
         expect(self.offerings.invokedPostOffer) == false
     }
 
+    func testGetPromotionalOfferFailsWithIneligibleIfReceiptHasNoTransactions() async throws {
```

Comment:
let's add another explicit test for the opposite case (doesn't fail if there are transactions). It's already indirectly tested, but having a direct one might help ensure that if other tests are refactored in some way, this one would still fail with a clear error